src/main/Scheduler.java

```diff
@@ -69,29 +71,31 @@ public void run() {
 					
 					for(DestinationEvent destEvent : destinationEvents) {
 						if(destEvent.getDestination() == arrivalEvent.getCurrentFloor()) {
-							keepGoing = !destinationEvents.isEmpty();
-							//set the floor event to null since thats what elevator wants. 
-							currentFloorEvent = null; 
 							destinationEventFlag = true; 
 							break; 
 						}
 					}
+					
+					//checks if the elevator should keep going. 
+					elevatorKeepsGoing = !destinationEvents.isEmpty();

```

Comment:
!destinationEvents.isEmpty() or you’re at a source floor then keepGoing = True

Reply:
done